### PR TITLE
SPLICE-2103: Remove Explain Generation from Hot Path for real-time que…

### PR DESCRIPTION
**Pull request by jleach4**
Remove Regex compilation from the execution of each node in the operation tree for real time queries.

### CODE REVIEW

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/SpliceBaseOperation.java [hunk 1]
```diff
@@ -187,10 +187,9 @@ public void setExplainPlan(String plan){
         // No difference. We can change that later if needed.
         // Right now this is only used by Spark UI, so don't change it
         // unless you want to change that UI.
-
         CompilerContext.DataSetProcessorType type = this.activation.getLanguageConnectionContext().getDataSetProcessorType();
-        // JL-TODO Cannot do this in the hot path
-        explainPlan=(plan==null?"":plan.replace("n=","RS=").replace("->","").trim());
+        if (type == CompilerContext.DataSetProcessorType.SPARK)
```
Comment (yxia92) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/SpliceBaseOperation.java]: I think we should add FORCED_SPARK too.

Comment (jleach4) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/SpliceBaseOperation.java]: Nice catch Yi...

Comment (jleach4) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/SpliceBaseOperation.java]: Pushed.

**Comment by jleach4** [comment]
"Hot Path" means that it runs for every operation in the tree for each execution.  The regex portions are not precompiled but have to be compiled each time as well.

**Comment by wtang16** [comment]
Current Spark UI has some issue displaying the explain plan on the nodes in DAG. We'll need to open a new JIRA for that, which is unrelated to this PR.

**Pull request merged**